gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +166,272 @@ def _get_prior_measurements_in_world_frame(
+    ) -> List[SfmTrack2d]:
+        """Removes bad tracks and selects the longest ones until all cameras see `measurements_per_camera` tracks.
+
+        Bad tracks are those that have fewer than 3 measurements from valid_cameras.
+        Sorts the remaining tracks in descending order, then selects the longest ones until all cameras see at least
+         `measurements_per_camera` tracks.
+        This is based on 1dsfm's implementation here:
+        https://github.com/wilsonkl/SfM_Init/blob/fd012ef93462b8623e8d65fa0c6fa95b32270a3c/sfminit/transproblem.py#L235
+
+        Args:
+            tracks: List of all input tracks.
+            valid_cameras: Set of valid camera indices (these have direction measurements and valid rotations).
+            intrinsics: List of camera intrinsics.
+            measurements_per_camera: Number of track direction measurements that need to be observed by each camera.
+
+        Returns:
+            List of tracks to use for averaging.
+        """
+        filtered_tracks = []
+        valid_cameras_with_intrinsics = set([c for c in valid_cameras if intrinsics[c] is not None])
+        for track in tracks:
+            valid_cameras_track = track.select_for_cameras(camera_idxs=valid_cameras_with_intrinsics)
+            if valid_cameras_track.number_measurements() < MIN_TRACK_MEASUREMENTS_FOR_AVERAGING:
+                continue
+            filtered_tracks.append(valid_cameras_track)
+
+        tracks_subset = []
+
+        remaining_cover = {c: measurements_per_camera for c in valid_cameras_with_intrinsics}
+        improvement = [t.number_measurements() for t in filtered_tracks]  # how much cover each track would add
+
+        # preparation: make a lookup from camera to tracks in the camera
+        camera_track_lookup = {c: [] for c in valid_cameras_with_intrinsics}
+
+        for track_id, track in enumerate(filtered_tracks):
+            for j in range(track.number_measurements()):
+                measurement = track.measurement(j)
+                camera_track_lookup[measurement.i].append(track_id)
+
+        for count in range(len(filtered_tracks)):  # artificial limit to avoid infinite loop
+            # choose track that maximizes the heuristic
+            best_track_id = np.argmax(improvement)
+            if improvement[best_track_id] <= 0:
+                break
+            tracks_subset.append(filtered_tracks[best_track_id])
+
+            # update the state variables: remaining_cover and improvement
+            improvement[best_track_id] = 0
+            for measurement in filtered_tracks[best_track_id].measurements:
+                # if this image just got covered the k'th time, it's done.
+                # lower the improvement for all tracks that see it
+                if remaining_cover[measurement.i] == 1:
+                    for t in camera_track_lookup[measurement.i]:
+                        improvement[t] = improvement[t] - 1 if improvement[t] > 0 else 0
```

Comment:
Nit: maybe move this step after the reduction in remaining cover, so that its more intuitive when checking against zero.